### ISSUE TITLE
as.TreeNumber not identifying unique topologies

### ISSUE THREAD

**Issue by jblandis** [ms609/TreeTools#142]
After going through the documentation, I was under the impression that as.TreeNumber would generate numbers for all the different unique topologies in a data set. Currently I am working with 299 trees consisting of seven tips (all trees have all tips). I can generate the output and save the scores for all 299 trees. When looking more closely at the trees with the same score, the topologies are in fact not the same. Am I misinterpreting the usage of this function or is there an error somewhere that is causing different topologies to be scored the same?

Here is the code that is being used:
trees <- read.tree(file="299_trees.tre")
trees

tips <- TipLabels(trees)
tips

possible <- as.TreeNumber(trees, nTip=7,tipLabels = tips)

possible
class(possible)

sink("output.txt",type=c("output"))

print(possible)
sink()

Thanks.

**Comment by ms609** [comment]
Hi, thanks for the query.  From a first glance, my guess would be that the trees' tip labels are not labelled in the same sequence in all trees, and thus that your value of `tips` is a list (rather than a character vector).  

`trees <- RenumberTips(trees, TipLabels(trees[[1]])` would ensure that the tips are labelled in a consistent order in all trees. If this doesn't solve your issue, then sending your tree file (as an attachment here, or by e-mail) would help me to explore further.

Either way it looks like I could improve the documentation, or adapt the function to support a list format for `tipLabels`, so please let me know how you get on.  

Thanks

Martin

**Comment by jblandis** [comment]
Hi Martin,

Thank you so much for your fast response. I tried two things today, the first was your suggestion with the tip labels and the second was I first sorted each tree by tip, and then used that new tip label variable:

tips <- TipLabels(trees)
sorted <- SortTree(trees, how = "tip")
tips2 <- TipLabels(sorted)
possible <- as.TreeNumber(sorted, nTip=7, tipLabels=tips2)

Scanning thru the results it seems like your suggestion and this sorting method give the exact same score for all 299 trees. In most cases this is the same as my first attempt, but in many places they do differ.

While I was trying to figure this out, I did find some code on the phytools blog that finds unique topologies in a set of trees and gives them an arbitrary number. I'm looking to see if that also corresponds well to what I just recovered. 

Thanks again for such a quick response.
